public/pages/AnomalyCharts/containers/AnomaliesChart.tsx

```diff
@@ -172,6 +177,17 @@ export const AnomaliesChart = React.memo((props: AnomaliesChartProps) => {
     );
   };
 
+  const hasValidHCProps = () => {
+    return (
+      props.isHCDetector &&
```

Comment:
Why we need to check so many props ? Can we use less props to check?

Reply:
It is to make sure user of `AnomaliesChart` doesn't miss needed props for HC detector, these props are optional because for non-HC detector, such props are not used. I can add comment for the condition around line 185 which can be not straightforward for understanding. Feel free to let me know your thoughts.

Reply:
Got it, please add some more comments for line 185-188